management/src/main/scala/akka/management/AkkaManagement.scala

```diff
@@ -0,0 +1,186 @@
+    _asyncAuthenticator = Option(auth)
+  // FIXME replace with config object and withs?
+  private[this] var _asyncAuthenticator: Option[AsyncAuthenticator[String]] = None
+
+  /**
+   * Set the HTTP(S) context that should be used when binding the management HTTP server.
+   *
+   * Set this to `akka.http.[javadsl|scaladsl].HttpsConnectionContext` to bind the server using HTTPS.
+   * Refer to the Akka HTTP documentation for details about configuring HTTPS for it.
+   */
+  def setHttpsContext(context: HttpsConnectionContext): Unit =
+    _connectionContext = context.asInstanceOf[akka.http.scaladsl.ConnectionContext]
+  private[this] var _connectionContext: akka.http.scaladsl.ConnectionContext = Http().defaultServerHttpContext
+
+  private val bindingFuture = new AtomicReference[Future[Http.ServerBinding]]()
+  private val selfUriPromise = Promise[Uri]() // TODO has to keep config as well as the Uri, so we can reject 2nd calls with diff uri
+
+  /**
+   * Start the HTTP management endpoint
+   */
+  // FIXME make it accept config object that would have all the `withHttps`
+  def start(): Future[Uri] = {
+    val serverBindingPromise = Promise[Http.ServerBinding]()
+    if (bindingFuture.compareAndSet(null, serverBindingPromise.future)) {
+
+      val hostname = settings.Http.Hostname
+      val port = settings.Http.Port
+
+      // port is on purpose never inferred from protocol, because this HTTP endpoint is not the "main" one for the app
+      val protocol = if (_connectionContext.isSecure) "https" else "http"
+      val selfBaseUri = Uri(s"$protocol://$hostname:$port/${settings.Http.BasePath}")
+      val providerSettings = ManagementRouteProviderSettingsImpl(selfBaseUri)
+
+      val combinedRoutes: Try[Route] = prepareCombinedRoutes(settings, providerSettings)
+
+      combinedRoutes match {
+        case Success(routes) ⇒
+          // TODO instead of binding to hardcoded things here, discovery could also be used for this binding!
+          // Basically: "give me the SRV host/port for the port called `akka-bootstrap`"
+          // discovery.lookup("_akka-bootstrap" + ".effective-name.default").find(myaddress)
+          // ----
+          // FIXME -- think about the style of how we want to make these available
+
+          val serverFutureBinding =
+            Http().bindAndHandle(
+              RouteResult.route2HandlerFlow(routes),
+              hostname,
+              port,
+              connectionContext = this._connectionContext
+            )
+
+          serverBindingPromise.completeWith(serverFutureBinding).future.flatMap { _ =>
+            log.info("Bound Akka Management (HTTP) endpoint to: {}:{}", hostname, port)
+            selfUriPromise.success(selfBaseUri).future
+          }
+
+        case Failure(ex) ⇒
+          log.warning(ex.getMessage)
+          Future.failed(new IllegalArgumentException("Failed to start Akka Management HTTP endpoint.", ex))
+      }
+    } else selfUriPromise.future
```

Comment:
I'd consider throwing if `start()` is called multiple times perfectly reasonable